frontend/src/main/components/MenuItemReview/MenuItemReviewForm.js

```diff
@@ -0,0 +1,157 @@
+import { Button, Form, Col } from 'react-bootstrap';
+import { useForm } from 'react-hook-form'
+import { useNavigate } from 'react-router-dom'
+
+function MenuItemReviewForm({ initialContents, submitAction, buttonLabel = "Create" }) {
+
+    // Stryker disable all
+    const {
+        register,
+        formState: {errors},
+        handleSubmit,
+    } = useForm(
+        {defaultValues: initialContents || {},}
+    );
+    // Stryker restore all
+
+    const navigate = useNavigate();
+
+    // For explanation, see: https://stackoverflow.com/questions/3143070/javascript-regex-iso-datetime
+    // Note that even this complex regex may still need some tweaks
+
+    // Stryker disable next-line Regex
+    const isodate_regex = /(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+)|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d)|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d)/i;
+
+    const testIdPrefix = "MenuItemReviewForm";
+
+    return(
+        <Form onSubmit={handleSubmit(submitAction)}>
+
+            {initialContents && (
+                <Col>
+                    <Form.Group className="mb-3" >
+                        <Form.Label htmlFor="id">Id</Form.Label>
+                        <Form.Control
+                            data-testid={testIdPrefix + "-id"}
+                            id="id"
+                            type="number"
```

Comment:
is number really the type we want here? People could submit something like this.
![image](https://github.com/ucsb-cs156-s24/team03-s24-4pm-3/assets/42985687/35d5ed98-a19f-46d6-a482-ab20244720c7)

Reply:
I think you would want to have it as text and validate using a regex instead, maybe something like [1-9][0-9]*   ?

Though this might break something else if it submits the value as a string instead of an integer. /shrug